### ISSUE TITLE
No image loads and 304 not modified

### ISSUE THREAD

**Issue by jbhanks** [openseadragon/openseadragon#2360]
I am having trouble getting images to display with OpenSeadragon. Logs indicate that the correct path is being found, as it finds the first couple tiles (but gives a 304 not modified). No controls are visible.

If I change the tileSources path to something that does not exist,  I get controls, but of course a 404 error as well).

```
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
    </head>
Under construction
<div id="seadragon-viewer" style="width: 100%; height: 99%;"></div>

<script src="static/openseadragon/openseadragon.min.js"></script><script type="text/javascript">
    var viewer = OpenSeadragon({
        id: "seadragon-viewer",
        prefixUrl: "static/openseadragon/images/",
        sequenceMode: true,
        showReferenceStrip: true,
        referenceStripScroll: 'horizontal',
        referenceStripSizeRatio: 0.1,
        maxZoomLevel: 2.0,
        tileSources: [
        "http://127.0.0.1:5002/static/dzi2/onion_tonecurves.dzi"
        ]
    });
</script>
{% include "footer.jinja2" %}
```

The output from my wsgi server:

```
127.0.0.1 - - [26/May/2023 04:23:23] "GET /static/openseadragon/images/next_grouphover.png HTTP/1.1" 304 -
127.0.0.1 - - [26/May/2023 04:23:23] "GET /static/openseadragon/images/next_pressed.png HTTP/1.1" 304 -
127.0.0.1 - - [26/May/2023 04:23:23] "GET /static/dzi2/onion_tonecurves.dzi HTTP/1.1" 304 -
127.0.0.1 - - [26/May/2023 04:23:23] "GET /static/dzi2/onion_tonecurves_files/2/0_0.jpeg HTTP/1.1" 304 -

```

Why might the image not be loading and is the 304 even related?

**Comment by iangilman** [comment]
I don't know... I'm not aware of there being an issue with 304 results. Is it possible for you to try giving the correct URL to avoid the 304, just to see if that makes a difference?

**Comment by jbhanks** [comment]
What do you mean by the "correct url"? If it found any of the files in subdirectories, I assume it to have gotten the correct url. Am I missing something?

**Comment by jbhanks** [comment]
Nevermind, I think I have it sorted out. I just didn't realize that I had to explicitly define a size for the viewer in the css style.

```
#openseadragon1 {
  width: 100vw;
  height: 100vh;
}
````

in my main.css did the trick.

**Comment by iangilman** [comment]
Oh yes... A common gotcha! Glad you got it sorted out. :)

**Comment by pearcetm** [comment]
> Oh yes... A common gotcha! Glad you got it sorted out. :)

Something for the to-be-created FAQ maybe? Or perhaps a viewer with unreasonably small actual dimensions could be detected and throw a console warning? I've been using OSD for years and this still trips me up sometimes when I'm using different CSS layouts than I'm used to!

**Comment by iangilman** [comment]
Yeah, an FAQ would probably be a good idea! Hmm... Looks like we already had an issue for that over in the site-build repository, but now I've made a new one here: https://github.com/openseadragon/openseadragon/issues/2368

But yeah, I think it would be even better if the app could warn you via the console, since not everyone is going to read the FAQ. That said, I'm concerned that in some cases, it would be normal operation for the viewer to have 0 size, for instance if it gets hidden sometimes. We'd want the warning to not happen in those cases. I'm not sure how best to detect that.

**Comment by iangilman** [comment]
Another good place to mention this would be on the starting page of the docs (https://openseadragon.github.io/docs/). The example there does include sizing for the div, but it's not explicitly called out.
